matchms/filtering/SpectrumProcessor.py

```diff
@@ -24,116 +23,106 @@ class SpectrumProcessor:
-        'fully_annotated', or None. Default is 'default'.
+    filters:
+        A list of filter functions, see add_filter for all the allowed formats.
     """
 
-    def __init__(self, predefined_pipeline: Optional[str] = 'default',
-                 additional_filters: Iterable[Union[str, List[dict]]] = ()):
+    def __init__(self,
+                 filters: Iterable[Union[Tuple[str, Dict[str, any]],
+                                         str,
+                                         Tuple[Callable, Dict[str, any]],
+                                         Callable]]):
         self.filters = []
         self.filter_order = [x.__name__ for x in ALL_FILTERS]
-        if predefined_pipeline is not None:
-            if not isinstance(predefined_pipeline, str):
-                raise ValueError("Predefined pipeline parameter should be a string")
-            if predefined_pipeline not in PREDEFINED_PIPELINES:
-                raise ValueError(f"Unknown processing pipeline '{predefined_pipeline}'. Available pipelines: {list(PREDEFINED_PIPELINES.keys())}")
-            for filter_name in PREDEFINED_PIPELINES[predefined_pipeline]:
-                self.add_matchms_filter(filter_name)
-        for filter_name in additional_filters:
-            if isinstance(filter_name, (tuple, list)) and len(filter_name) == 1:
-                filter_name = filter_name[0]
-            self.add_filter(filter_name)
-
-    def add_filter(self,
-                   filter_function: Union[Tuple[str, Dict[str, any]], str, Tuple[Callable, Dict[str, any]], Callable]):
-        """Add a filter to the processing pipeline. Takes both matchms filter names (and parameters)
-        as well as custom-made functions.
-        """
-        if isinstance(filter_function, str):
-            self.add_matchms_filter(filter_function)
-        elif isinstance(filter_function, (tuple, list)) and isinstance(filter_function[0], str):
-            self.add_matchms_filter(filter_function)
-        else:
-            self.add_custom_filter(filter_function[0], filter_function[1])
-
-    def add_matchms_filter(self, filter_spec: Union[Tuple[str, Dict[str, any]], str]):
-        """
-        Add a filter to the processing pipeline.
-
-        Parameters
-        ----------
-        filter_spec : str or tuple
-            Name of the filter function to add, or a tuple where the first element is the name of the
-            filter function and the second element is a dictionary containing additional arguments for the function.
+        for filter_name in filters:
+            self.parse_and_add_filter(filter_name)
+
+    def parse_and_add_filter(self, filter_description: Union[Tuple[str, Dict[str, any]],
+                                                             str,
+                                                             Tuple[Callable, Dict[str, any]],
+                                                             Callable],
```

Comment:
Maybe it would be worth to look into typedefs for these things: https://stackoverflow.com/questions/69446189/python-equivalent-for-typedef